components/context/src/main/java/datadog/context/WeakMapContextBinder.java

```diff
@@ -1,29 +1,36 @@
 package datadog.context;
 
-import java.util.Collections;
+import static datadog.context.Context.root;
+import static java.util.Collections.synchronizedMap;
+import static java.util.Objects.requireNonNull;
+
 import java.util.Map;
 import java.util.WeakHashMap;
+import javax.annotation.ParametersAreNonnullByDefault;
 
 /** {@link ContextBinder} that uses a global weak map of carriers to contexts. */
+@ParametersAreNonnullByDefault
 final class WeakMapContextBinder implements ContextBinder {
-
-  private static final Map<Object, Context> TRACKED =
-      Collections.synchronizedMap(new WeakHashMap<>());
+  private static final Map<Object, Context> TRACKED = synchronizedMap(new WeakHashMap<>());
 
   @Override
   public Context from(Object carrier) {
+    requireNonNull(carrier, "Context carrier can't be null");
```

Comment:
```suggestion
    requireNonNull(carrier, "Context carrier cannot be null");
```
I prefer to avoid things like `'` in log messages, as they can complicate parsing/grepping :)

Reply:
Make sense. I am never sure of which form is the "right" one to use but log parsing is definitely a valid point.